Sources/Core/Animation/Common/UIKit/Extension/UIView+AnimationExtension.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  UIView+AnimationExtension.swift
+//  SparkCommon
+//
+//  Created by robin.lemaire on 18/11/2024.
+//  Copyright © 2024 Adevinta. All rights reserved.
+//
+
+import UIKit
+
+public extension UIView {
+
+    /// Animate a view with a Spark animation.
+    /// - parameter type: the spark animation type.
+    /// - parameter view: the view to animate.
+    /// - parameter delay: the delay before before starting the animation. Default is *.zero*.
+    /// - parameter repeat: the repeat type of the animation. Default is *.once*.
+    /// - parameter completion: the completion when the animation is finished. *Optional*. Default is *nil*.
+    class func animate(
+        withType type: SparkAnimationType,
```

Comment:
Type being strongly type you can either animate(with: type), animate(type: type) or animate(type)